src/main/java/io/stargate/sgv2/jsonapi/service/embedding/DataVectorizer.java

```diff
@@ -49,6 +51,9 @@ public DataVectorizer(
     this.nodeFactory = nodeFactory;
     this.embeddingCredentials = embeddingCredentials;
     this.schemaObject = schemaObject;
+    // This is getting element at 0 since only one vector is stored in the schema.
+    // This logic needs to be changed to handle multiple vectors columns for tables,
+    vectorConfig = schemaObject.vectorConfigs().get(0);
```

Comment:
Should class Javadoc probably mention if it is only being used for Collections?
I assume it is because only single vector is defined as per comment here.

Reply:
This class has all field names hard coded for collections, this needs to be adapted for api tables.